### ISSUE TITLE
New Resource: aws_ec2_local_gateway_route_table_vpc_association

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#13501]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Support the ability to manage EC2 Local Gateway Route Table VPC Associations, connecting Outposts to VPCs.

### New Resource

* aws_ec2_local_gateway_route_table_vpc_association

### Potential Terraform Configuration

```hcl
resource "aws_ec2_local_gateway_route_table_vpc_association" "example" {
  local_gateway_route_table_id = data.aws_ec2_local_gateway_route_table.example.id
  vpc_id                       = aws_vpc.example.id
}
```

### References

- [Original Issue](https://github.com/terraform-providers/terraform-provider-aws/issues/12302)
- [EC2 CreateLocalGatewayRouteTableVpcAssociation API Reference](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateLocalGatewayRouteTableVpcAssociation.html)
- [Outposts Documentation](https://docs.aws.amazon.com/outposts/latest/userguide/outposts-local-gateways.html#vpc-associations)

**Comment by FireballDWF** [comment]
@bflad  If you want help testing this on an actual Outpost, send me a test build of the provider and I can likely turn around results in a day

**Comment by bflad** [comment]
@FireballDWF very much appreciate the offer! It looks like we'll have access to an AWS Outposts lab again June 3rd-5th, but it would be nice to get ahead of that. 😄

**Comment by bflad** [comment]
Hey @FireballDWF / @martinsmatthews -- code can now be found at https://github.com/terraform-providers/terraform-provider-aws/pull/13765. Please reach out to me at my username at hashicorp dot com if you need me to provide you a build directly (with which OS/arch), otherwise we have instructions available in the [Development Guide](https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/DEVELOPMENT.md#quick-start). Thanks!

**Comment by bflad** [comment]
Support for this new resource has been merged and will release with version 2.67.0 of the Terraform AWS Provider, likely tomorrow. 👍
